addons/udes_stock/models/res_users.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, SUPERUSER_ID, _
+from odoo.exceptions import ValidationError
+
+
+class ResUser(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def get_user_warehouse(self, aux_domain=None):
+        """Get the warehouse(s) of the user by chain of the company
+        :kwargs:
+            - aux_domain:
+                If specified must return a single warehouse, if want a subset
+                of warehouses then don't specify the aux_domain and filter result
+        :returns: Warehouse(s), or a singular warehouse if aux_domain not None
+        """
+        Warehouse = self.env["stock.warehouse"]
+
+        user_id = self.env.uid
+
+        if user_id != SUPERUSER_ID:
+            user = self.search([("id", "=", user_id)])
+            if not user:
+                raise ValidationError(_("Cannot find user"))
+
+        domain = [("company_id", "=", self.env.user.company_id.id)]
```

Comment:
the commit in 13.0 that added this:
```
    [FIX] udes_stock: Added check for system user in get_user_warehouse
     
    A check is carried out in get_user_warehouse to see if the user is active. However the system user is excluded from searches. An earlier check is now carried out to see if the user_id matches SUPERUSER_ID. If it doesn't then the check is carried out as before.
     
    Story/9859
     
    Signed-off-by: Peter Clark <peter.clark@unipart.io>
```
i don't know if the self.env.uid / self.env.user is relevant, there's no comment to say otherwise so i'll change it

Reply:
I wouldn't trust that guy. 👀

I'm not sure what the suggested change is.

The `user_id != SUPERUSER_ID` was added to avoid unit tests falling over as the system user was being reported as not found on v13. :)

Reply:
basically using `self.env.user` and `self.env.user.id` only, rather than switching between `self.env.user` and `self.env.uid`